server/team/team.route.js

```diff
@@ -5,9 +5,20 @@ const { ifNoUserRedirect } = require('../middlewares/user');
 const { ifNoBody400 } = require('../middlewares/util');
 const teamCtrl = require('./team.contoller');
 const validators = require('./team.validation');
+const User = require('../user/user.model');
 
 const router = express.Router(); // eslint-disable-line new-cap
 
+function deleteValidator(req, res, next) {
+  return User.findOne({ _id: req.user._id })
+  .then((user) => {
+    if (user.teamId !== req.params.teamId) {
+      return next(new Error('Unauthorized'));
```

Comment:
We should only throw an error if it is something unexpected. There is plenty of opportunity for sneaky people to try and delete an opponents team, so we expect that and give them a 400 in return.